src/components/error-boundary.tsx

```diff
@@ -0,0 +1,68 @@
+import { Alert, AlertTitle, Stack, Typography } from '@mui/material';
+
+import React from 'react';
+
+interface IErrorBoundaryProps {
+  alertTitle: string;
+  alertMessage: string;
+  detailTitle: string;
+}
+
+interface IErrorBoundaryState {
+  hasError: boolean;
+  error?: unknown;
+}
+
+export class ErrorBoundary extends React.Component<
+  React.PropsWithChildren<IErrorBoundaryProps>,
+  IErrorBoundaryState
+> {
+  constructor(props: IErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(error: unknown): IErrorBoundaryState {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true, error: error };
+  }
+
+  componentDidCatch(error: unknown, errorInfo: React.ErrorInfo): void {
+    // You can also log the error to an error reporting service
+    // errorInfo has full stack trace, which we are not using
+    this.setState({ hasError: true, error });
+  }
+
+  render(): React.ReactNode {
+    let errorDetail;
+    if (typeof this.state.error === 'string') {
+      errorDetail = this.state.error;
+    } else if (this.state.error instanceof Error) {
+      errorDetail = this.state.error.message;
+    }
```

Comment:
The problem here is that if the raised error is not an Error object or a string, then the error detail doesn't get shown. Since all objects contain a `toString()` method in JS, you should be able to do:

```tsx
const errorDetail = this.state.error instanceof Error
  ? this.state.error.message
  : this.state.error.toString();
```

Reply:
Obviously, seeing `[Object object]` would not be particularly helpful, but it would be better than nothing.

Reply:
Unfortunately, `unknown.toString()` doesn't pass `tsc`

Reply:
Hmm, looks like `toString()` can't be called on `null` or `undefined`. I think the safest case is to keep the current implementation rather than risk triggering a second exception and rendering a blank screen when catching one.